pandas/core/arrays/string_.py

```diff
@@ -279,11 +268,27 @@ def __setitem__(self, key, value):
                 value = np.asarray(value, dtype=object)
             if len(value) and not lib.is_string_array(value, skipna=True):
                 raise ValueError("Must provide strings.")
+        return value
+
+    def __setitem__(self, key, value):
```

Comment:
this is weird github formatting - `__setitem__` wasn't touched. Rather I moved the validation portion of the method to `_validate_setitem_value` and call that